src/esp/gfx/Simulator.cpp

```diff
@@ -350,5 +350,21 @@ double Simulator::getWorldTime() {
   return NO_TIME;
 }
 
+std::shared_ptr<nav::PathFinder> Simulator::recomputeNavMesh(
```

Comment:
Use a unique_ptr here. The caller can convert to shared_ptr if they want to .

https://herbsutter.com/2013/05/29/gotw-89-solution-smart-pointers/

Reply:
With re-design this is no longer an issue. :)